### PR TITLE
Make additional repo holder optional for 4.3 /proxy. Change print to logger

### PR DESCRIPTION
**Pull request by maximenoel8**
## Context

We added back proxy additional repository holder in main.tf for 5.0.
We don't need / use this place holder for 4.3 so far. 

## What does this  PR change ?

Make the place holder optional. 
Change print by logger for more clarity.

### CODE REVIEW

File: terracumber/terraformer.py [hunk 1]
```diff
@@ -85,8 +85,10 @@ def inject_repos(self, custom_repositories_json):
                         (new_line, n) = subn(placeholder, replacement, line)
                         print(new_line, end='')
                         n_replaced += n
-                    if n_replaced != 1:
+                    if n_replaced > 1:
                         return 2
+                    elif n_replaced == 0:
+                        return 3
```
Comment (NamelessOne91) [inline terracumber/terraformer.py:91]: The function docstring state that return codes are 0 , 1  and 2.
I don't know if returning 3 here might be a problem or we don't really care (eg: we only check if the return is 0 or not when we call this function).

Just pointing it out :)

Comment (NamelessOne91) [reply terracumber/terraformer.py:91]: Nevermind, I see you also added a check for when the return is 3.
All that is missing then is just mentioning it in the doctstring.

**Review by NamelessOne91 — APPROVED** [review]
LGTM

Thanks Maxime

**Review by ktsamis — APPROVED** [review]

**Comment by NamelessOne91** [comment]
Merged ASAP as this is needed to properly restart 4.3 BV

File: terracumber/terraformer.py (hunk 1 above)
Comment (srbarrios) [reply terracumber/terraformer.py:91]: In the future, it would be nicer to have proper Error classes raised at that point. Make sense?

**Pull request merged**